action.php

```diff
@@ -691,8 +691,18 @@ protected function page_depend_replacements($raw, $id) {
         $replace['@PAGEURL@'] = wl($id, $params, true, "&");
         $replace['@QRCODE@']  = $qr_code;
 
+        $content = '';
+
+        // let other plugins define their own replacements
+        $evdata = ['replace' => &$replace, 'content' => &$content];
```

Comment:
I would pass `$id` in the event data, too.

Reply:
good idea, done